python/arcticdb/diagnostics_engine/library_options.py

```diff
@@ -0,0 +1,286 @@
+    """,
+    "symbol_list": """
+    \n If enabled, the symbol list caching functionality of Arctic will be used. This is designed to improve
+    performance of list_symbols operations. For a more detailed understanding of this feature, including how it
+    works and its various use cases and limitations, please refer to the specific 'Symbol list consistency
+    diagnosis' section in this notebook.""",
+}
+
+
+class LibraryInfoGetter:
+    """
+    This class provides a method for displaying and explaining the various options
+    associated with a specific library. It aims to provide users with a clear understanding
+    of each option, including its purpose, usage implications, and potential impact on
+    library operations.
+
+    Methods:
+    --------
+    display_library_options():
+        Fetches and displays the options for a given library, along with detailed
+        explanations for each option.
+
+    Example Usage:
+    --------------
+    info_getter = LibraryInfoGetter(lib, EXPLANATIONS)
+    info_getter.display_library_options()
+    """
+
+    def __init__(self, lib, explanations):
+        self.response_obj = lib._lib_cfg.lib_desc.version
+        self.explanations = explanations
+
+        self.response_non_default_str = (
+            "## Next, see the list of your library options that has been modified with respect to the default"
+            " values:\n\n"
+        )
+        self.response_default_str = (
+            "## Next, see the list of your library options that remains with the default values:\n\n"
+        )
+
+        # Default values defined internally
+        self.internal_default_values = {
+            "column_group_size": 127,
+            "segment_row_size": 100000,
+            "max_num_buckets": 150,
+        }
+
+    def traverse_descriptor(self, obj):
+        """
+        Traverses the fields of the provided protobuf object, compares their values with default ones,
+        and updates response strings accordingly. The function is recursive for nested protobuf message types.
+        """
+        default_response_obj = type(obj)()
+        for field in obj.DESCRIPTOR.fields:
+            value = getattr(obj, field.name)
+            if field.message_type:
+                self.traverse_descriptor(value)
+            elif field.name in self.explanations:
+                default_value = getattr(default_response_obj, field.name)
+                explanation = self.explanations.get(field.name, "No explanation required.")
+                if value != default_value:
+                    if field.name in self.internal_default_values:
+                        default_value = self.internal_default_values[field.name]
+                    self.response_non_default_str += (
+                        f"\n### **{field.name}**: {value} (Default: {default_value})\n\n{explanation}\n\n"
+                    )
+                else:
+                    if field.name in self.internal_default_values:
```

Comment:
No, I can't do that. I've already considered this possibility. If the first check (line 241) passes, we set the default_value variable. If the second check (line 247) passes, we set the value variable. Note that options defined internally within the library typically have their default value set to '0'. Therefore, we need to preserve this default_value until it can be compared with the actual value. If it has changed, we restore the actual default value (as defined internally). If it hasn't changed, then we know the actual value is the one defined internally.

I acknowledge it's a bit off. But it's primarily due to the fact that we hardcode things internally, which I believe is the root of the problem.

Reply:
Oh I see what you mean